doc/_scripts/gen_boards_catalog.py

```diff
@@ -70,15 +70,15 @@ def get_catalog():
         arch_roots=module_settings["arch_root"],
         board_roots=module_settings["board_root"],
         soc_roots=module_settings["soc_root"],
-        board_dir=ZEPHYR_BASE / "boards",
+        board_dir=[ZEPHYR_BASE / "boards"],
```

Comment:
```suggestion
        board_dir=[],
```

this seems to do the trick. No idea why. Also please consider renaming to `board_dirs` if this is now an array

Reply:
> it looks like I may have misunderstood what "board_dir" is for

board dir is intended to indicate locations where to look for boards.
This is useful during incremental builds where the board dirs are cached by CMake and thus we can pass the exact directories to `list_boards.py` and thereby avoid searching all folders for a given board.

The reason we still invoke list_boards even when we know the location is to be able to both detect renames, folder movement, changes in board variants, etc., and there the list_board.py has all the knowledge.

Will change to `[]`, as that seems to be what you actually want in this case.

Regarding rename to `board_dirs` then you have point when the list_boards is imported into another Python, but as `--board-dir` is actually a command line argument, which can be given multiple times, like `list_boards.py --board-dir=<path>/foo-board --board-dir=<path>/bar-board --board-dir=...`, then I actually think the use of singular in the name in this case is correct.

Reply:
changed to `[]`.